### DISCUSSION TITLE
How we can uno all the last 3 4 query’s were execute on database to undo the values change in tables

### DISCUSSION THREAD

**Discussion by sadaqatdev** [simolus3/drift#2524] [General]
How we can uno all the last 3 4 query’s were execute on database to undo the values change in tables

**Comment by simolus3** [comment]
There's no builtin undo functionality in sqlite3 that drift could expose.

If you anticipate the need to undo queries, you could run them in a [transaction](https://drift.simonbinder.eu/docs/transactions/). If that doesn't work, for instance because you need to undo them much later, you'd have to manually keep a "log" table that stores changes and a way to undo them.
